src/runtime/internal/style_manager.ts

```diff
@@ -31,7 +31,7 @@ export function create_rule(node: Element & ElementCSSInlineStyle, a: number, b:
 	const name = `__svelte_${hash(rule)}_${uid}`;
 	const doc = node.ownerDocument as ExtendedDoc;
 	active_docs.add(doc);
-	const stylesheet = doc.__svelte_stylesheet || (doc.__svelte_stylesheet = doc.head.appendChild(element('style') as HTMLStyleElement).sheet as CSSStyleSheet);
+	const stylesheet = doc.__svelte_stylesheet || (doc.__svelte_stylesheet = append_empty_stylesheet().sheet as CSSStyleSheet);
```

Comment:
You will have some issue there: all animation are instantiated later than the component.

So it means that, if you have multiple shadowRoot using some svelte components you will have some trouble with animations

Reply:
can you explain why this code would be an issue?

Reply:
I'm not 100% sure of what I'm saying there 😌

But for me the create_rule function is not call during the lifecycle of the dom creation.

so let's imagine this situation:

```typescript
const container1 = document.createElement('div');
const container2 = document.createElement('div');

container1.attachShadow({ mode: 'open' });
container2.attachShadow({ mode: 'open' });

new SvelteComponentA({
   target: container1.shadowRoot,
   styleTarget: container1.shadowRoot,
});

new SvelteComponentB({
   target: container2.shadowRoot,
   styleTarget: container2.shadowRoot,
});
```

for me the issue is that: if the SvelteComponentA is using some animations, all the css will be injected inside _container2.shadowRoot_ instead of _container2.shadowRoot_

Reply:
ok, now I understand your problem. I will take a look